### PR TITLE
use Homebrew.jl to grab cfitsio on OSX

### PR DESCRIPTION
**Pull request by mweastwood**
I don't have easy access to a Mac so this is untested. Hopefully Travis comes back with a green light.

This *might* fix some of the issues reported in mweastwood/LibHealpix.jl#38, but I think this is a good idea regardless.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,4 @@
 julia 0.5
 BinDeps 0.3.12
 Compat 0.17.0
+@osx Homebrew
```
Comment (kbarbary) [inline REQUIRE]: Huh, did not know this was legal syntax in REQUIRE!

**Comment by kbarbary** [comment]
Do you know, if an OSX user *doesn't* have homebrew, will build-from-source still be used?

**Comment by mweastwood** [comment]
Homebrew.jl is using is doing its own thing. It doesn't install things globally using the system Homebrew installation so I don't believe having Homebrew is a pre-req for this (Homebrew.jl is though). If this is merged, I think it will always be preferred to the build-from-source method on OSX.

On the plus side though, it means other packages that need libcfitsio (ie. LibHealpix.jl) can use the same installation.

**Comment by kbarbary** [comment]
I see. Seems OK to me, but I'm not on a mac. I guess we can always change it if people have problems using Homebrew.jl. Does anyone with a mac have an opinion?

File: deps/build.jl [hunk 2]
```diff
@@ -62,6 +62,12 @@ if is_windows()
     push!(BinDeps.defaults, BuildProcess)
 end
 
+# OSX
+if is_apple()
```
Comment (giordano) [inline deps/build.jl]: Add
```julia
using Compat.Sys: iswindows, isapple, isunix
```
after the `using Compat` line and replace all `is_<os>` with `is<os>`.  You should also rise `Compat` version to 0.29.0 in `REQUIRE`.

Comment (mweastwood) [inline deps/build.jl]: updated

**Pull request merged**